### ISSUE TITLE
Provide a way to pretty-print Value

### ISSUE THREAD

**Issue by radeksimko** [zclconf/go-cty#34]
While `cty.Value`'s `GoString()` is useful for basic debugging when one is already focusing on a simple types/values. More often in Terraform though we need to deal with deeply nested structures, where such one-lined output is not very readable.

It would therefore be useful to have a separate function, which (similarly) prints out values in a human-readable format, but accounts for more complex structures by indenting the output.

### Example

This is the current output from `GoString()`:

`cty.ObjectVal(map[string]cty.Value{"allow":cty.SetVal([]cty.Value{cty.ObjectVal(map[string]cty.Value{"ports":cty.UnknownVal(cty.List(cty.String)), "protocol":cty.UnknownVal(cty.String)})}), "creation_timestamp":cty.NullVal(cty.String), "deny":cty.SetVal([]cty.Value{cty.ObjectVal(map[string]cty.Value{"ports":cty.UnknownVal(cty.List(cty.String)), "protocol":cty.UnknownVal(cty.String)})}), "description":cty.NullVal(cty.String), "destination_ranges":cty.NullVal(cty.Set(cty.String)), "direction":cty.NullVal(cty.String), "disabled":cty.NullVal(cty.Bool), "enable_logging":cty.NullVal(cty.Bool), "id":cty.NullVal(cty.String), "name":cty.UnknownVal(cty.String), "network":cty.UnknownVal(cty.String), "priority":cty.NullVal(cty.Number), "project":cty.NullVal(cty.String), "self_link":cty.NullVal(cty.String), "source_ranges":cty.NullVal(cty.Set(cty.String)), "source_service_accounts":cty.NullVal(cty.Set(cty.String)), "source_tags":cty.NullVal(cty.Set(cty.String)), "target_service_accounts":cty.NullVal(cty.Set(cty.String)), "target_tags":cty.NullVal(cty.Set(cty.String)), "timeouts":cty.NullVal(cty.Object(map[string]cty.Type{"create":cty.String, "delete":cty.String, "update":cty.String}))})`

which would be much more readable in format similar to this:

```go
cty.ObjectVal(map[string]cty.Value{
	"allow": cty.SetVal([]cty.Value{
		cty.ObjectVal(map[string]cty.Value{
			"ports":    cty.UnknownVal(cty.List(cty.String)),
			"protocol": cty.UnknownVal(cty.String),
		}),
	}),
	"creation_timestamp": cty.NullVal(cty.String),
	"deny":cty.SetVal([]cty.Value{
		cty.ObjectVal(map[string]cty.Value{
			"ports":    cty.UnknownVal(cty.List(cty.String)),
			"protocol": cty.UnknownVal(cty.String),
		}),
	}),
	"description":			cty.NullVal(cty.String),
	"destination_ranges":		cty.NullVal(cty.Set(cty.String)),
	"direction":			cty.NullVal(cty.String),
	"disabled":			cty.NullVal(cty.Bool),
	"enable_logging":		cty.NullVal(cty.Bool),
	"id":				cty.NullVal(cty.String),
	"name":				cty.UnknownVal(cty.String),
	"network":			cty.UnknownVal(cty.String),
	"priority":			cty.NullVal(cty.Number),
	"project":			cty.NullVal(cty.String),
	"self_link":			cty.NullVal(cty.String),
	"source_ranges":		cty.NullVal(cty.Set(cty.String)),
	"source_service_accounts":	cty.NullVal(cty.Set(cty.String)),
	"source_tags":			cty.NullVal(cty.Set(cty.String)),
	"target_service_accounts":	cty.NullVal(cty.Set(cty.String)),
	"target_tags":			cty.NullVal(cty.Set(cty.String)),
	"timeouts":			cty.NullVal(cty.Object(map[string]cty.Type{
		"create": cty.String,
		"delete": cty.String,
		"update": cty.String,
	}),
)})
```

**Comment by pselle** [comment]
Maybe we can add a `FriendlyString` a la `FriendlyName` for types https://godoc.org/github.com/zclconf/go-cty/cty#Type.FriendlyName

**Comment by apparentlymart** [comment]
In the past when I've found myself looking at complicated types and values I've pretty-printed them using [`dump.Value`](https://pkg.go.dev/github.com/apparentlymart/go-dump@v0.0.0-20190214190832-042adf3cf4a0/dump?tab=doc#Value), which is a best-effort pretty-printer for `GoString` values. It's not perfect but in my experience it's done a good enough job to help me debug in a pinch.

**Comment by apparentlymart** [comment]
I've pushed up a separate package [`ctydebug`](https://godoc.org/github.com/zclconf/go-cty-debug/ctydebug) which contains some utilities of this sort, including one to pretty-print values in a manner similar to what you suggested here.

I've made it separate because I expect I'll want to tweak the exact output of those functions over time based on experience using them and it's easier to iterate quickly without having to make a new `cty` release every time. I don't expect to be changing the Go API for that package, but the exact formatting of the strings the functions return might change in future.
